### PR TITLE
List disk types

### LINKED ISSUE #51166: Disk types not listed
Looking at https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree#table_engine-mergetree-multiple-volumes_configure there are no docs on the full list of supported `type`s. I know there is `s3` from the S3 docs, and there seem to also be `web` and `cache` based on examples of nested dynamic storage, but it would be very helpful to know the list of types and how to use them.

### PR DESCRIPTION
**Pull request by DanRoscigno**
closes https://github.com/ClickHouse/ClickHouse/issues/51166

In the referenced issue we were asked for the list of disk types and how to use them.  

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit bfcadabb927e5ea547c29df488d3fe6ea396a178 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51539/bfcadabb927e5ea547c29df488d3fe6ea396a178/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 1]
```diff
@@ -756,6 +756,15 @@ If you perform the `SELECT` query between merges, you may get expired data. To a
 - [ttl_only_drop_parts](/docs/en/operations/settings/settings.md/#ttl_only_drop_parts) setting
 
 
+## Disk types
+
+In addition to local block devices, ClickHouse supports other device types through table engines. These are the types:
```
Comment (alexey-milovidov) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:761]: This statement is subtly incorrect.
Disk abstraction is different from integration table engines.

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 2]
```diff
@@ -756,6 +756,15 @@ If you perform the `SELECT` query between merges, you may get expired data. To a
...
+- [S3](#table_engine-mergetree-s3)
+- GCS (also supported using the [S3 table engine](#table_engine-mergetree-s3))
+- [Azure Blob Storage](#table_engine-mergetree-azure-blob-storage)
+- [HDFS](#hdfs-storage)
+- [Web (read-only)](#web-storage)
```
Comment (alexey-milovidov) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:766]: Let's name them exactly as in the configuration file to avoid confusion.

Comment (alexey-milovidov) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:766]: Also, cache and s3_plain.

**Comment by danthegoodman1** [comment]
👏

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 3]
```diff
@@ -756,6 +756,17 @@ If you perform the `SELECT` query between merges, you may get expired data. To a
...
+## Disk types
+
+In addition to local block devices, ClickHouse supports these storage types:
+- [`s3` for S3 and MinIO](#table_engine-mergetree-s3)
+- [`gcs` for GCS](/docs/en/integrations/data-ingestion/gcs/index.md/#creating-a-disk)
+- [`blob_storage_disk` for Azure Blob Storage](#table_engine-mergetree-azure-blob-storage)
+- [`hdfs` for HDFS](#hdfs-storage)
+- [`web` for read-only from web](#web-storage)
+- [`cache` for local caching](/docs/en/operations/storing-data.md/#using-local-cache)
+- [`s3_plain` for backups to S3](/docs/en/operations/backup#backuprestore-using-an-s3-disk)
```
Comment (kssenii) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:768]: btw there is also local_blob_storage disk type, but it is a toy feature, so not important to document.

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 4]
```diff
@@ -1238,6 +1251,93 @@ Examples of working configurations can be found in integration tests directory (
...
+## Web storage (read-only) {#web-storage}
+
+Web storage can be used for read-only purposes. An example use is for hosting sample
+data, or for migrating data.
+
+:::tip
+Storage can also be configured temporarily within a query, if a web dataset is not expected
+to be used routinely, see [dynamic storage](#dynamic-storage) and skip editing the
+configuration file.
+:::
```
Comment (kssenii) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:1299]: there is a program in clickhouse named `static-files-disk-uploader` which helps to generate files for disk web.

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 5]
```diff
@@ -1238,6 +1251,93 @@ Examples of working configurations can be found in integration tests directory (
...
+
+In this sample configuration:
+- the disk is of type `web`
+- the data is hosted at `http://nginx:80/test1/`
```
Comment (kssenii) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:1303]: a better example from clickhouse-public-datasets https://pastila.nl/?003f0701/1e981fdc7770d010afdf02882ac40813

**Pull request merged**